### PR TITLE
fix: input focus styles

### PR DESCRIPTION
**Pull request by alva-fresh**
https://dev.azure.com/EY-GA/Global%20Atlantic%20DevOps/_workitems/edit/23661

## Description

- bug fix - `:focus` state for inputs within the `Form.Field`;
- Also, fixed the same issue for `textarea`;

## Motivation

To match the designs by indicating the selected input;

## Screenshots

Screenshot 2023-06-14 at 1 20 31 PM

## Testing

Run the app -> make sure that the border color changes to green;

### CODE REVIEW

File: src/styles/input.css [hunk 1]
```diff
@@ -7,7 +7,9 @@
   font-size: var(--font-size);
 }
 
-.ui.input > input:focus {
+.ui.input > input:focus,
+.ui.form input[type='text']:focus,
+.ui.form textarea:focus {
```
Comment (michael-lynch) [inline src/styles/input.css:23]: Nice catch on the `textarea` but let's add _all_ of the other Semantic selectors as well. I'm not sure why Semantic is styled like this, but I figure we should match it to cover our bases.

```
.ui.form input:not([type]):focus,
.ui.form input[type="date"]:focus,
.ui.form input[type="datetime-local"]:focus,
.ui.form input[type="email"]:focus,
.ui.form input[type="number"]:focus,
.ui.form input[type="password"]:focus,
.ui.form input[type="search"]:focus,
.ui.form input[type="tel"]:focus,
.ui.form input[type="time"]:focus,
.ui.form input[type="text"]:focus,
.ui.form input[type="file"]:focus,
.ui.form input[type="url"]:focus,
.ui.form textarea:focus {
  ...
}
```

File: src/components/pages/Theme/Theme.jsx [hunk 2]
```diff
@@ -273,10 +294,20 @@ export const Theme = () => {
               placeholder='Select options'
             />
 
+            {inputTypes.map((type) => (
```
Comment (michael-lynch) [inline src/components/pages/Theme/Theme.jsx]: Let's move these into the `<Input>` segment and display a grid of inputs.

File: src/components/pages/Theme/Theme.jsx [hunk 3]
```diff
@@ -273,10 +294,20 @@ export const Theme = () => {
...
+                label={type}
+                type={type}
+              />
+            ))}
+
             <Form.Field
               control={TextArea}
-              label='About'
-              placeholder='Tell us more about you...'
+              label='Textarea'
```
Comment (michael-lynch) [inline src/components/pages/Theme/Theme.jsx]: Let's create a separate segment for `TextArea`.

**Review by michael-lynch — CHANGES_REQUESTED** [review]

File: src/components/pages/Theme/Theme.jsx (hunk 3 above)
Comment (alva-fresh) [reply src/components/pages/Theme/Theme.jsx]: Should it be within the `Input` segment?

**Comment by Dart-EY** [comment]
Maybe for a future ticket, we should also fix the styling of the text input for the `error` variant:

When used in a label:

**CURRENT**
image


**DESIGN**
image

File: src/components/pages/Theme/Theme.jsx [hunk 4]
```diff
@@ -248,6 +263,9 @@ export const Theme = () => {
               <Input id='input-default' placeholder='Input' />
             </Form.Field>
 
+            <Header size='small' dividing>
+              Sates
```
Comment (michael-lynch) [inline src/components/pages/Theme/Theme.jsx]: "States"

File: src/components/pages/Theme/Theme.jsx [hunk 5]
```diff
@@ -300,6 +302,31 @@ export const Theme = () => {
               )
             }
           />
+
+          <Header size='small' dividing>
+            Types
+          </Header>
+          <Form>
+            <Form.Field
```
Comment (michael-lynch) [inline src/components/pages/Theme/Theme.jsx]: Let's remove the Dropdown from this segment, as it has its own segment.

**Review by michael-lynch — CHANGES_REQUESTED** [review]

**Comment by michael-lynch** [comment]
> Maybe for a future ticket, we should also fix the styling of the text input for the `error` variant:

@alva-fresh Do you want to include this in this PR or would you prefer we create a separate ticket?

File: src/styles/form.css [hunk 6]
```diff
@@ -13,6 +13,11 @@
   text-transform: none;
 }
 
+/* Focus */
+.ui.form .field:focus-within > label {
+  color: var(--primary-pms-362-900);
```
Comment (michael-lynch) [inline src/styles/form.css:18]: This will color the label green on focus.

**Review by michael-lynch — APPROVED** [review]
I added a new `.grid` class and used that to lay out the different inputs. I think it just makes the layout a little cleaner.

File: src/styles/form.css (hunk 6 above)
Comment (alva-fresh) [reply src/styles/form.css:18]: Thank you!!

**Pull request merged**